recipe/build.sh

```diff
@@ -5,10 +5,18 @@ rm -rf build || true
 
 
 if [ ${cuda_compiler_version} != "None" ]; then
-  CUDA_CMAKE_OPTIONS="-DCMAKE_CUDA_COMPILER=${CUDA_HOME}/bin/nvcc -DCMAKE_CUDA_HOST_COMPILER=${CXX} -DCUDA_ARCH_NAME=All "
+  CUDA_CMAKE_OPTIONS="-DCUDA_ARCH_NAME=All"
   USE_CUDA=ON
   # Remove -std=c++17 from CXXFLAGS for compatibility with nvcc
   CXXFLAGS="$(echo $CXXFLAGS | sed -e 's/ -std=[^ ]*//')"
+
+  # Add NVVM's `bin` directory to the `$PATH`.
+  # This should fix an error finding `cicc`.
+  # ref: https://forums.developer.nvidia.com/t/when-i-arch-option-error-sh-cicc-command-not-found-takes-place/31753/2
+  # xref: https://github.com/conda-forge/cuda-nvcc-impl-feedstock/issues/9
+  if [[ "${cuda_compiler_version}" == 12* ]]; then
+    export PATH="${PATH}:${BUILD_PREFIX}/nvvm/bin"
+  fi
```

Comment:
Was seeing the following error [on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=874043&view=logs&j=2ed0550f-e734-5ebf-52c6-1bfc19f5811c ) (also [attached log]( https://github.com/conda-forge/dgl-feedstock/files/14217764/dgl-feedstock_20240208.18_linux_64_c_compiler_version12cuda_compilercuda-nvcccuda_compiler_version12.0cxx_compiler_version12python3.10.____cpython.txt ) for posterity)

```
sh: cicc: command not found
CMake Error at gpu_cache_generated_nv_gpu_cache.cu.o.Release.cmake:276 (message):
  Error generating file
  /home/conda/feedstock_root/build_artifacts/dgl_1707443486363/work/build/CMakeFiles/gpu_cache.dir/third_party/HugeCTR/gpu_cache/src/./gpu_cache_generated_nv_gpu_cache.cu.o
```

As a workaround, followed the advice in [this thread]( https://forums.developer.nvidia.com/t/when-i-arch-option-error-sh-cicc-command-not-found-takes-place/31753/2 ) to add it to the `$PATH`. It is worth noting that TensorFlow ran into a similar issue: https://github.com/conda-forge/cuda-nvcc-impl-feedstock/issues/9